File: baekjoon/silver/1308/1308.py
# input
# 2008 12 27
# 2009 1 22
# output
# D-26

import datetime

leap_year = [31,29,31,30,31,30,31,31,30,31,30,31]
norm_year = [31,28,31,30,31,30,31,31,30,31,30,31]

for i in range(1, 12):
  leap_year[i] = leap_year[i] + leap_year[i-1]
  norm_year[i] = norm_year[i] + norm_year[i-1]

s_y, s_m, s_d = map(int, input().split())
e_y, e_m, e_d = map(int, input().split())

s_dt = datetime.datetime(s_y, s_m, s_d)
e_dt = datetime.datetime(e_y, e_m, e_d)

ans = 0

def is_leap_year(year):
  if year % 400 == 0:
    return True
  elif year % 100 == 0:
    return False
  elif year % 4 == 0:
    return True
  else:
    return False

if datetime.datetime(s_y + 1000, s_m, s_d) <= e_dt:
  print('gg')
else:
  for i in range(s_y, e_y):
    if is_leap_year(i):
      ans += 366
    else:
      ans += 365
  if s_m != 1:
    if is_leap_year(s_y):
      ans -= leap_year[s_m - 2]
    else:
      ans -= norm_year[s_m - 2]
  if e_m != 1:
    if is_leap_year(e_y) and e_m:
      ans += leap_year[e_m - 2]
    else:
      ans += norm_year[e_m - 2]
  ans -= s_d
  ans += e_d
  print(f'D-{ans}')

File: baekjoon/silver/1308/1308.go
package main

import "fmt"

func main() {
	var leapYear []int = []int{31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}
	var normYear []int = []int{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}

	var sy, sm, sd, ey, em, ed, ans int
	fmt.Scanf("%d %d %d", &sy, &sm, &sd)
	fmt.Scanf("%d %d %d", &ey, &em, &ed)

	for i := 1; i < 12; i++ {
		leapYear[i] += leapYear[i-1]
		normYear[i] += normYear[i-1]
	}

	if isGG(sy, sm, sd, ey, em, ed) {
		fmt.Println("gg")
		return
	}

	for i := sy; i < ey; i++ {
		if isLeapYear(i) {
			ans += 366
		} else {
			ans += 365
		}
	}
	if sm != 1 {
		if isLeapYear(sy) {
			ans -= leapYear[sm-2]
		} else {
			ans -= normYear[sm-2]
		}
	}
	if em != 1 {
		if isLeapYear(ey) {
			ans += leapYear[em-2]
		} else {
			ans += normYear[em-2]
		}
	}
	fmt.Printf("D-%v\n", ans-sd+ed)
}

func isLeapYear(year int) bool {
	return year%4 == 0 && year%100 != 0 || year%400 == 0
}

func isGG(sy, sm, sd, ey, em, ed int) bool {
	if sy+1000 < ey {
		return true
	} else if sy+1000 == ey && sm < em {
		return true
	} else if sy+1000 == ey && sm == em && sd <= ed {
		return true
	}
	return false
}
